src/posix_regex.h

```diff
@@ -7,34 +7,45 @@
 #ifndef INCLUDE_posix_regex_h__
 #define INCLUDE_posix_regex_h__
 
-#include <regex.h>
-
 /*
  * Regular expressions: if the operating system has p_regcomp_l,
- * use that as our p_regcomp implementation, otherwise fall back
- * to standard regcomp.
+ * use it so that we can override the locale environment variable.
+ * Otherwise, use our bundled PCRE implementation.
  */
 
+#ifdef GIT_USE_REGCOMP_L
+# include <regex.h>
+# include <xlocale.h>
+
 #define P_REG_EXTENDED REG_EXTENDED
 #define P_REG_ICASE REG_ICASE
 #define P_REG_NOMATCH REG_NOMATCH
 
 #define p_regex_t regex_t
 #define p_regmatch_t regmatch_t
 
-#define p_regerror regerror
-#define p_regexec regexec
-#define p_regfree regfree
-
-#ifdef GIT_USE_REGCOMP_L
-#include <xlocale.h>
-
 GIT_INLINE(int) p_regcomp(p_regex_t *preg, const char *pattern, int cflags)
 {
 	return regcomp_l(preg, pattern, cflags, (locale_t) 0);
 }
+
+#define p_regerror regerror
+#define p_regexec regexec
+#define p_regfree regfree
+
 #else
-# define p_regcomp regcomp
+# include "pcreposix.h"
```

Comment:
I think it might make sense to restructure our compatibility files a bit. Maybe have a "compat" or "posix" directory that contains all platform-specific abstractions, including our "unix" and "win32" directories as well as these new ones.

Edit: oh, this comes directly from the PCRE library. Didn't get that at first